### ISSUE TITLE
Extract the development docs from this repository

### ISSUE THREAD

**Issue by tacaswell** [matplotlib/matplotlib#22866]
We have historically built all of our documentation from the sphinx source in the main matplotlib/matplotlib repo (this one).  We would then host this by unpacking the result of the build at the top level of website + into a version specific sub-folder.  We would also rely on 

While this served us well for over a decade, over the past few years we have slowly started to pull things apart.  The goal is to ultimately de-couple the version independent parts of the documentation (which should be re-published as soon as they are update) from the version-specific docs (which should be    Steps we have already taken so far:

 - move the cannonical docs to https://matplotlib.org/stable https://discourse.matplotlib.org/t/canonical-documentation-have-moved/21863
 - split the third-party docs out into its own repo https://github.com/matplotlib/mpl-third-party
 - have a sphinx theme for all of our websites https://github.com/matplotlib/mpl-sphinx-theme
 - split the top-level index into its own repo https://github.com/matplotlib/mpl-brochure-site
 - have built docs for the governance https://github.com/matplotlib/governance
 - moved our ground-truth hosting to digital ocean https://github.com/matplotlib/matplotlib.org

The next thing we should pull out into its own repository is the content under https://matplotlib.org/stable/devel/index.html .  We probably should copy-paste the GHA/sphinx configuration from the governance repo and update the rst in that framework.

We should do this as a second step after we finish @noatamir 's update of that content (to make it easier to review the changes at each step).

**Comment by QuLogic** [comment]
Do we want to put this in `mpl-brochure-site` or a separate repository?

**Comment by tacaswell** [comment]
Originally I thought `mpl-brocure-site`, but I now I think a separate repository.  We already have a whole bunch of repos serving docs so we might as well lean into that and separate the repos by purpose.

**Comment by timhoffm** [comment]
> Originally I thought `mpl-brocure-site`, but I now I think a separate repository. We already have a whole bunch of repos serving docs so we might as well lean into that and separate the repos by purpose.

Makes sense. It does not fit into `mpl-brochure-site`.

We may need a place to document which repositories are used for which parts of the docs. - Should this be in a separate repo as well? :stuck_out_tongue_winking_eye:

**Comment by story645** [comment]
> An overview of the structure/architecture of the docs would fit in the docs no?!

Could maybe go in the dev guide for doc contributions since "where is this thing I want to change?" is a not infrequent new contributor question.

**Comment by tacaswell** [comment]
Currently we are pulling the artifacts for our docs from 14 different repos (see 
https://github.com/matplotlib/matplotlib.org/blob/e57c891aaddf8fda1c5fd06dcb5c99875e2c05bf/matplotlib.org.yml#L10-L24) so although I agree with the sentiment, I think we are well past hoping to keep the number of repositories involved to very small numbers.  The very large number of repos contributing to the docs is not new, we were previously exploiting the implicit layering that github pages so it was much harder to see!

If we focus just on the core library docs we have:

 - matplotlib.github.com : this is the built output of our versioned docs + cmocean docs.  Everything in here should be strictly output of build scripts.  We should not really do PRs (maybe should just turn them off?) and probably want to lock down write access to the release managers?
 - devdocs : this is built and published on merge to main.  This should never get PRs, the push process force-pushes so we only ever have one commit (😱).  We should probably disable PRs on this repo and only leave write access to the deploy key?
 - goverance : in addition to holding the governance this now has a build and publish pipeline for a rendered version of the governance.  This repo should be pared down to just be the governance (process and people)+  its build pipeline and write access restricted to the steering council
 - mpl-brochure : this is just the top-level index.html + its build pipeline. This probably should also have write permissions restricted to a small set of people (release manager + ?)
 - cheatsheets : There are (as the name says) the cheatsheets. It has its own build process and Nicolas is responsible person for this repo, we should defer to him about what the write permissions should be
 - matplotlblog: this should probably be replaced with a re-direct to the https://blog.scientific-python.org/tags/matplotlib/ at the caddy level and archived (that is a separate issue though)
 - mpl-third-party : the third-party page which has a very custom build (like cheatsheets).  I'll defer to @jklymak about what we want the write permissions on this to be
 - (proposed) mpl-devel-guide : the content being proposed to be pulled out here.  I think we want the write permissions on this to exactly match the write permissions on the main repo.  This is both to keep the level of write access the same as it is now and because I think that everyone we trust to merge code to the library we should trust their judgement to document how to work on it!

I've been thinking about security for unrelated reasons recently.  I am not concerned about what people we know and have given access to are going to do, however I am concerned about what someone who gets control of one of our accounts might do which is why I want to restrict write permissions on the goverance and very repos that are the source for the high-traffic parts of our website.

I do not think there is anything we really want to merge together, we are already paying the complexity cost of "more that one" at the caddy configuration level, and wanting to have different review/merge rules on https://matplotlib.org and the development guide I think we should go with (just) one more repo and include the above in the devel guide as a part of how to work on the docs.

**Comment by jklymak** [comment]
> and wanting to have different review/merge rules on https://matplotlib.org/ and the development guide I think we should go with (just) one more repo and include the above in the devel guide as a part of how to work on the docs.

OK, that makes sense.  maybe just leave it open to be more than just "dev guide" in case we want other docs not tied explicitly to the release cycle?

**Comment by tacaswell** [comment]
> OK, that makes sense. maybe just leave it open to be more than just "dev guide" 

That seems correct to me, but no better term is coming to mind quickly.  I would also want to move the communication guidelines out of governance to the new repo along with guides/templates on how to handle common situations with (new) contributors ("we do not assign issues, just start working", "congrats on your first PR, hope to here from you again!", "this better handled on discourse", "can you please include a minimal running example + version numbers", "your local installation is broken, uninstall everything and try using just 1 (one) package manager", etc), the triage guide, the PR review guide, etc, the meta-doc on docs, etc.

In our usual way, https://matplotlib.org/stable/devel/index.html is `devel` in the source tree, "Develop" in the site header, and has an H1 of "Contributing" (with a nested H2 that is _also_ "Contributing").

"contributing-and-maintaining-and-community-guide" is too long.

**Comment by jklymak** [comment]
I don't think we need to move anything - simply point `/stable/devel` to `/devdocs/devel`, etc for other things that should be current.  _Or_ if we don't want folks popped into `devdocs`, just automatically rsync the subdirectory into `/stable` whenever `/devdocs` is updated.  When this last came up, I don't think we had the possibility of server side redirects.

**Comment by jklymak** [comment]
Ping @QuLogic for an opinion on if either of the above are doable.

**Comment by QuLogic** [comment]
Here is the followup PR from the meeting discussion: https://github.com/matplotlib/matplotlib.org/pull/36
